src/telemetry.js

```diff
@@ -0,0 +1,94 @@
+var _ = require('./utility');
+
+function Telemeter(options) {
+  this.queue = [];
+  this.options = _.extend(true, {}, options);
+  var maxTelemetryEvents = this.options.maxTelemetryEvents;
+  // pin the queue size to the interval [0, 100]
+  this.maxQueueSize = Math.max(0, Math.min(maxTelemetryEvents || 100, 100))
+}
+
+Telemeter.prototype.copyEvents = function() {
+  return Array.prototype.slice.call(this.queue, 0);
+};
+
+Telemeter.prototype.capture = function(type, metadata, level) {
+  var e = {
+    level: getLevel(type, level),
+    type: type,
+    timestamp_ms: _.now(),
+    body: metadata
+  };
+  this.push(e);
+  return e;
+};
+
+Telemeter.prototype.captureEvent = function(metadata, level) {
+  return this.capture('manual', metadata, level);
+}
+
+Telemeter.prototype.captureError = function(err) {
+  return this.capture('error', {
+    message: err.message || String(err)
+  }, 'error');
+};
+
+Telemeter.prototype.captureLog = function(message, level) {
+  return this.capture('log', {
+    message: message
+  }, level);
+};
+
+Telemeter.prototype.captureNetwork = function(metadata, subtype) {
+  subtype = subtype || 'xhr';
+  metadata.subtype = metadata.subtype || subtype;
+  return this.capture('network', metadata, 'info');
```

Comment:
If this is an XHR event, can we get the status code and change the level based on it?